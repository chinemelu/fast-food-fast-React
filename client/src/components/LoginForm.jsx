import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import TextFieldGroup from './TextFieldGroup';
import FlashMessageList from './FlashMessagesList';

const LoginForm = ({
  loginErrors,
  email,
  password,
  onChange,
  onBlur,
  onClickForgotPasswordLink,
  onInput,
  onSubmit,
  isLoading
}) => (
  <section id="login-section">
    <form className="cd-form" autoComplete="off" onSubmit={onSubmit}>
      <FlashMessageList
        customAlertClass="login-alert"
      />
      <TextFieldGroup
        error={loginErrors && loginErrors.email}
        labelValue="Email"
        labelClass="input-label"
        labelFor="login-username-email"
        field="email"
        value={email}
        onChange={onChange}
        onBlur={onBlur}
        feedbackClass="login-error-message"
        className="input-field"
        id="login-email"
        inputContainerClass="input-container"
        onInput={onInput}
      />

      <TextFieldGroup
        error={loginErrors && loginErrors.password}
        type="password"
        labelValue="Password"
        labelClass="input-label"
        labelFor="login-password"
        field="password"
        value={password}
        onChange={onChange}
        onBlur={onBlur}
        id="login-password"
        feedbackClass="login-error-message"
        className="input-field"
        inputContainerClass="input-container"
        onInput={onInput}
      />

      <TextFieldGroup
        type="checkbox"
        labelValue="Remember me"
        feedbackClass=""
        className="input-field"
        id="remember-me"
        inputContainerClass="input-container"
      />

      <TextFieldGroup
        type="submit"
        value="LOGIN"
        feedbackClass=""
        className="submit-form-btn"
        id="login-form-btn"
        inputContainerClass="input-container"
        disabled={isLoading || Object.keys(loginErrors).length > 0}
      />
      <Link
        id="forgot-password"
        to="#"
        onClick={onClickForgotPasswordLink}
      >
          Forgot password?
      </Link>
    </form>
  </section>
);

LoginForm.propTypes = {
  email: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  loginErrors: PropTypes.objectOf(PropTypes.string).isRequired,
  onChange: PropTypes.func.isRequired,
  onBlur: PropTypes.func.isRequired
};

export default LoginForm;
