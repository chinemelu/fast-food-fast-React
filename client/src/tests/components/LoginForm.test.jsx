import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from '../../components/LoginForm';

describe('Render Login Page', () => {
  describe('render Login Page', () => {
    const props = {
      onChange: () => {},
      onBlur: () => {},
      onInput: () => {},
      loginErrors: 'some error',
      onSubmit: () => {},
      email: 'some@email.com',
      password: 'password',
      isLoading: false,
      onClickForgotPasswordLink: jest.fn()
    };
    it('renders the login page component', () => {
      const wrapper = shallow(<LoginForm {...props} />);
      expect(wrapper.find('TextFieldGroup').length).toEqual(4);
      expect(wrapper.find('Link').length).toEqual(1);
      expect(wrapper.find('Link').length).toEqual(1);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
