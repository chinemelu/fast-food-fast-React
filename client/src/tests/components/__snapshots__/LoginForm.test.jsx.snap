// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Login Page render Login Page renders the login page component 1`] = `
<section
  id="login-section"
>
  <form
    autoComplete="off"
    className="cd-form"
    onSubmit={[Function]}
  >
    <Connect(FlashMessageList)
      customAlertClass="login-alert"
    />
    <TextFieldGroup
      className="input-field"
      disabled={null}
      error=""
      feedbackClass="login-error-message"
      field="email"
      id="login-email"
      inputContainerClass="input-container"
      labelClass="input-label"
      labelFor="login-username-email"
      labelValue="Email"
      onBlur={[Function]}
      onChange={[Function]}
      onInput={[Function]}
      placeholder=""
      type="text"
      value="some@email.com"
    />
    <TextFieldGroup
      className="input-field"
      disabled={null}
      error=""
      feedbackClass="login-error-message"
      field="password"
      id="login-password"
      inputContainerClass="input-container"
      labelClass="input-label"
      labelFor="login-password"
      labelValue="Password"
      onBlur={[Function]}
      onChange={[Function]}
      onInput={[Function]}
      placeholder=""
      type="password"
      value="password"
    />
    <TextFieldGroup
      className="input-field"
      disabled={null}
      error=""
      feedbackClass=""
      id="remember-me"
      inputContainerClass="input-container"
      labelValue="Remember me"
      onBlur={null}
      placeholder=""
      type="checkbox"
    />
    <TextFieldGroup
      className="submit-form-btn"
      disabled={true}
      error=""
      feedbackClass=""
      id="login-form-btn"
      inputContainerClass="input-container"
      onBlur={null}
      placeholder=""
      type="submit"
      value="LOGIN"
    />
    <Link
      id="forgot-password"
      onClick={[MockFunction]}
      replace={false}
      to="#"
    >
      Forgot password?
    </Link>
  </form>
</section>
`;
